File: Medium/2026-07-16-3867-Sum-of-GCD-of-Formed-Pairs/solution.js
function gcd(a, b) {
    while (b) {
        [a, b] = [b, a % b];
    }
    return a;
}

function sumGCD(nums) {
    const n = nums.length;
    const prefixGcd = new Array(n);
    let maxNum = 0;

    for (let i = 0; i < n; i++) {
        maxNum = Math.max(maxNum, nums[i]);
        prefixGcd[i] = gcd(nums[i], maxNum);
    }

    prefixGcd.sort((a, b) => a - b);
    let sum = 0;

    for (let i = 0; i < Math.floor(n / 2); i++) {
        sum += gcd(prefixGcd[i], prefixGcd[n - 1 - i]);
    }

    return sum;
}

File: Medium/2026-07-16-3867-Sum-of-GCD-of-Formed-Pairs/solution.java
import java.util.Arrays;

public class Solution {
    public int sumOfGcdOfFormedPairs(int[] nums) {
        int n = nums.length;
        int[] prefixGcd = new int[n];
        int max = nums[0];
        
        for (int i = 0; i < n; i++) {
            max = Math.max(max, nums[i]);
            prefixGcd[i] = gcd(nums[i], max);
        }
        
        Arrays.sort(prefixGcd);
        int sum = 0;
        
        for (int i = 0; i < n / 2; i++) {
            sum += gcd(prefixGcd[i], prefixGcd[n - 1 - i]);
        }
        
        return sum;
    }
    
    private int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}